services/datamanager/builtin/builtin.go

```diff
@@ -63,8 +62,8 @@ const defaultCaptureBufferSize = 4096
 // Attributes to initialize the collector for a component or remote.
 type dataCaptureConfig struct {
 	Name               string               `json:"name"`
-	Model              string               `json:"model"`
-	Type               resource.SubtypeName `json:"type"`
+	Model              resource.Model       `json:"model"`
+	Type               resource.Subtype     `json:"type"`
```

Comment:
For quick background: we create a collector for each unique combination of [component name/model/type and method name/parameters]. 

* To create a collector on a component, we grab its nested data_manager config for the capture frequency and method name, and set the component name/model/type to that of the component config.
* To create a collector on a remote, we grab its nested data_manager config for the capture frequency, method name, and set the component name/type and remote robot name to that of the remote config.

This works well with modular resources by extracting the type into a string like `acme:demo:mygizmo`, but want to make sure there's parity between components and resources and is obvious what the user should input: e.g. should `Type` in ResourceLevelServiceConfig actually be named `API`?

Reply:
Actually, the more I look at this, I realize data_manager is the ONLY service using ResourceLevelServiceConfig so far. I'm going to undo this part of things for now, just go back to the SubtypeName as the "Type" field. In one or more of those later PRs, this should eventually get upgraded to triplets to support modular service_config services (it'll likely need to have a model as well.) I will notify when done so maybe you can take a final look.

Reply:
Since we're already expecting a full triplet (e.g. rdk:component:arm/myRemoteArm) as the type for remotes and parse the name/type from it, this should be backwards compatible and work with Subtype as the "Type" field.

I believe the only thing I'd change here to not break existing data capture configs would be to set `attrs.Type = c.Type` since they won't have the API field below (`attrs.Type = c.API`).

Reply:
And +1 to keeping ResourceLevelServiceConfig as-is — right now, the type is always just "data_manager" but that will change as we can now (woo!) easily implement different types of data managers.

Reply:
Regarding backward compatible... I can't directly use c.Type since it's only a subtypename, and c.API _should_ be filled out by any loaded config during the Validate() stage... but to be cautious, I'm changing it to `attrs.Type = c.ResourceName().Subtype` because the ResourceName() call will automatically generate the API field if it wasn't set, so should be safe in either case.